src/reducers/AuthReducer.js

```diff
@@ -24,16 +47,46 @@ export default (state = INITIAL_STATE, action) => {
             return ({...state, appReady: true});
 
         case GUEST_TOKEN_SET:
-            return ({...state, token: payload.token, appReady: payload.appReady});
+            const guestUser = {name: payload.token};
+            return ({...state, token: payload.token, currentUser: guestUser, appReady: payload.appReady});
 
         case FACEBOOK_LOGIN_SUCCESS:
-            return ({...state, token: payload.token, appReady: payload.appReady});
+            return ({...state, token: payload.token, currentUser: payload.currentUser, appReady: payload.appReady});
 
         case FACEBOOK_LOGIN_FAILURE:
             return ({...state, token: null});
 
         case FACEBOOK_LOGOUT_COMPLETE:
-            return ({...state, token: null, appReady: false});
+            return (INITIAL_STATE);
+
+        case FAVORITE_ADDED:
+            const {favoriteToAdd} = payload;
+
+            let updatedUserWithNewFavorite = _.cloneDeep(state.currentUser);
+            updatedUserWithNewFavorite.favorites = _.concat(updatedUserWithNewFavorite.favorites, favoriteToAdd);
+
+            return ({...state, currentUser: updatedUserWithNewFavorite});
+
+        case FAVORITE_REMOVED:
+            const {favoriteToRemove} = payload;
+            let updatedUserWithoutFavorite = _.cloneDeep(state.currentUser);
+            updatedUserWithoutFavorite.favorites = _.reject(updatedUserWithoutFavorite.favorites, favorite => favorite.id === favoriteToRemove.id);
+
+            return ({...state, currentUser: updatedUserWithoutFavorite});
+
+        case INITIALIZE_MAP:
+            const {sites} = payload;
+
+            if (state.currentUser && state.currentUser.name !== tokens.GUEST) {
+                const hydratedFavorites = hydrateFavorites({currentUser: state.currentUser, sites});
+                let userWithHydratedFavorites = _.cloneDeep(state.currentUser);
+                userWithHydratedFavorites.favorites = hydratedFavorites;
+
+                return ({...state, currentUser: userWithHydratedFavorites});
+            } else {
+                return state;
```

Comment:
As we saw in more detail in the `hydrateFavorites` function... whenever we have sites come in, we pass them to `hydrateFavorites` and either leave them as-is if they're already full favorites, or finish building them out if it's our first time logging in.